### PR TITLE
Adding checks for edisp, psf and bkg in MapDatasetEventSampler.run()

### PR DESCRIPTION
**Pull request by fabiopintore**
This PR introduces a check for the existence of either the psf or edisp or bkg in the dataset object in the `MapDatasetEventSampler.run` method in `gammapy.cube`.

### CODE REVIEW

File: gammapy/cube/simulate.py [hunk 1]
```diff
@@ -321,27 +321,43 @@ def event_list_meta(dataset, observation):
...
-        observation : `~gammapy.data.Observation`
-            In memory observation.
-        edisp : Bool
-            It allows to include or exclude the Edisp in the simulation.
-
-        Returns
-        -------
-        events : `~gammapy.data.EventList`
-            Event list.
+            Parameters
```
Comment (adonath) [inline gammapy/cube/simulate.py]: Please fix the indentation here...

File: gammapy/cube/simulate.py [hunk 2]
```diff
@@ -321,27 +321,43 @@ def event_list_meta(dataset, observation):
...
         """
-        events_bkg = self.sample_background(dataset)
         events_src = self.sample_sources(dataset)
-        events_src = self.sample_psf(dataset.psf, events_src)
 
-        events_src = self.sample_edisp(dataset.edisp, events_src)
+        if dataset.psf:
+            events_src = self.sample_psf(dataset.psf, events_src)
+        else:
+            events_src.table.add_columns(
```
Comment (adonath) [inline gammapy/cube/simulate.py]: Maybe just use:
```
event_src.table["RA"] = event_src.table["RA_TRUE"]
event_src.table["DEC"] = event_src.table["DEC_TRUE"]
```
Seems easier to read...

File: gammapy/cube/simulate.py [hunk 3]
```diff
@@ -321,27 +321,43 @@ def event_list_meta(dataset, observation):
...
+                (events_src.table["RA_TRUE"].data, events_src.table["DEC_TRUE"].data),
+                names=("RA", "DEC"),
+            )
+
+        if dataset.edisp:
+            events_src = self.sample_edisp(dataset.edisp, events_src)
+        else:
+            events_src.table.add_column(
```
Comment (adonath) [inline gammapy/cube/simulate.py]: See my comment above...

File: gammapy/cube/tests/test_simulate.py [hunk 4]
```diff
@@ -182,10 +182,40 @@ def test_mde_run(dataset):
     sampler = MapDatasetEventSampler(random_state=0)
     events = sampler.run(dataset=dataset, observation=obs)
 
-    assert len(events.table) == 2349
-    assert_allclose(events.table["ENERGY"][0], 1.894698, rtol=1e-5)
-    assert_allclose(events.table["RA"][0], 266.454448, rtol=1e-5)
-    assert_allclose(events.table["DEC"][0], -30.870316, rtol=1e-5)
+    assert len(events.table) == 2422
```
Comment (adonath) [inline gammapy/cube/tests/test_simulate.py]: Why did those test values change? It seems unrelated somehow...

File: gammapy/cube/simulate.py (hunk 1 above)
Comment (fabiopintore) [reply gammapy/cube/simulate.py]: OK.

File: gammapy/cube/simulate.py (hunk 2 above)
Comment (fabiopintore) [reply gammapy/cube/simulate.py]: OK

File: gammapy/cube/simulate.py (hunk 3 above)
Comment (fabiopintore) [reply gammapy/cube/simulate.py]: OK

File: gammapy/cube/tests/test_simulate.py (hunk 4 above)
Comment (fabiopintore) [reply gammapy/cube/tests/test_simulate.py]: Yeah I know, it's the same thing I was asking myself. They shouldn't change but they did.

File: gammapy/cube/tests/test_simulate.py [hunk 5]
```diff
@@ -182,10 +182,40 @@ def test_mde_run(dataset):
...
+    assert_allclose(events.table["DEC"][0], -28.45051813404372, rtol=1e-5)
+
+    meta = events.table.meta
+
+    assert meta["RA_PNT"] == 266.4049882865447
+    assert meta["ONTIME"] == 36000.0
+    assert meta["OBS_ID"] == 1001
+    assert meta["RADECSYS"] == "icrs"
+
+
```
Comment (adonath) [inline gammapy/cube/tests/test_simulate.py]: Please add the missing `requires_data()` decorator here...

File: gammapy/cube/tests/test_simulate.py (hunk 4 above)
Comment (adonath) [reply gammapy/cube/tests/test_simulate.py]: I think they changed because the order in which the sample methods are called changed. So the seed for the random call, might be a different one...
